src/Repository/RepositoryBase.php

```diff
@@ -70,24 +72,24 @@ public function findOneByAuthor($author): ?Post
         return $this->findOne(compact('author'));
     }
 
-    public function findAllByAuthor($author): null|PostCollectionInterface
+    public function findAllByAuthor($author): Collection
     {
         if (!$author) {
-            return null;
+            return new Collection([]);
         }
         if (is_object($author)) $author = $author->ID;
         return $this->find(compact('author'));
     }
 
-    public function findAllDrafts(): ?PostCollectionInterface
+    public function findAllDrafts(): Collection
     {
         return $this->find(['posts_per_page' => -1, 'post_status' => 'draft']);
     }
 
-    public function findWithIds(array $post_ids, int $count = 10): null|PostCollectionInterface
+    public function findWithIds(array $post_ids, int $count = 10): Collection
     {
         if (empty($post_ids)) {
-            return null;
+            return new Collection([]);
```

Comment:
I'm guessing with a Decorator pattern, it'll be better to handle empty cases internally in Collection with a zero-item PostQuery or something.